.github/workflows/publish.yml

```diff
@@ -2,16 +2,61 @@ name: publish
 
 on:
   release:
-    types: [published]
+    types: [published] # Trigger when a GitHub Release is published
+
+env:
+  DOCS_REPO: SAP/ai-sdk
+  NODE_VERSION: 20
 
 jobs:
-  publish:
+  publish-npm:
+    name: Publish to NPM
     runs-on: ubuntu-latest
-
+    outputs:
+      version: ${{ steps.get_version.outputs.version }}
     steps:
       - uses: sap/ai-sdk-js/.github/actions/setup@main
+        with:
+          node-version: ${{ env.NODE_VERSION }}
       - name: publish
         run: |
-          yarn changeset publish
+          pnpm changeset publish
         env:
           NODE_AUTH_TOKEN: ${{ secrets.NPMJS_ACCESS_TOKEN }}
+      - name: Get Package Version
+        id: get_version
+        run: |
+          VERSION=$(node -p "require('./package.json').version")
+          echo "Using version: $VERSION"
+          echo "version=$VERSION" >> "$GITHUB_OUTPUT"
+
+  merge-release-notes-pr:
+    name: Merge Release Notes PR
+    needs: [publish-npm]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: sap/ai-sdk-js/.github/actions/setup@main
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+      - name: Determine Docs PR Branch Name
+        id: determine-branch-name
+        run: |
+          VERSION=${{ needs.publish-npm.outputs.version }}
```

Comment:
[req] Can't we check version in this job?

Reply:
Done.